server/tserver/src/main/java/org/apache/accumulo/tserver/TabletClientHandler.java

```diff
@@ -968,9 +969,19 @@ public List<TCMResult> conditionalUpdate(TInfo tinfo, long sessID,
       return results;
     } catch (IOException ioe) {
       throw new TException(ioe);
+    } catch (Exception e) {
+      log.warn("Exception returned for conditionalUpdate {}", e);
+      errorThrown = true;
+      throw e;
     } finally {
-      writeTracker.finishWrite(opid);
-      server.sessionManager.unreserveSession(sessID);
+      try {
+        writeTracker.finishWrite(opid);
+        server.sessionManager.unreserveSession(sessID);
+      } catch (Exception e) {
+        var message = errorThrown ? "Second Exception" : "Exception";
```

Comment:
Is this block really needed?  This is presuming that we can thrown an exception from finishWrite or unreserveSession.  If we can thrown an exception from the former, then this should be split into 2 finally blocks.  If not, then perhaps this block is not needed.  It appears that we are guarenteed that the writeTracker.startWrite was called and returned a valid opid.  In addition the sessID must have been reserved.  The only way these would throw exceptions is if some other thread for the same opid or sessID and I do not see that as being possible.

Reply:
Yeah you're right. While the sessionID is passed in to `conditionalUpdate`, the session is reserved on line 926 and the write opid is also generated on line 944. 
I agree that it should be impossible for another thread to have the same opid or reserved sessID at the same time, so the exceptions from finishWrite and unreserveSession wouldn't happen here. 

I've gone ahead and pulled this block out.